Source/WebCore/cssjit/SelectorCompiler.cpp

```diff
@@ -3380,11 +3344,13 @@ void SelectorCodeGenerator::generateElementAttributeMatching(Assembler::JumpList
 
             const AtomStringImpl* namespaceURI = attributeSelector.attribute().namespaceURI().impl();
             if (namespaceURI) {
-                LocalRegister namespaceToMatch(m_registerAllocator);
-                m_assembler.move(Assembler::TrustedImmPtr(namespaceURI), namespaceToMatch);
-                successCases.append(m_assembler.branchPtr(Assembler::Equal, Assembler::Address(qualifiedNameImpl, QualifiedName::QualifiedNameImpl::namespaceMemoryOffset()), namespaceToMatch));
+                if (attributeSelector.attribute().nodeNamespace() != Namespace::Unknown) {
+                    static_assert(sizeof(Namespace) == sizeof(uint8_t));
+                    successCases.append(m_assembler.branch8(Assembler::Equal, Assembler::Address(qualifiedNameImpl, QualifiedName::QualifiedNameImpl::namespaceMemoryOffset()), Assembler::TrustedImm32(static_cast<uint8_t>(attributeSelector.attribute().nodeNamespace()))));
```

Comment:
Because `branch8` relies on that (not branch16, branch32, for example). So. we are explicitly putting static_assert and use `uint8_t` which is what the assembler want to see here.